[release-2.1 < T0769-MG] Add retry transaction docs (#244)
memgraph/docs: docs/reference-guide/streams/overview.md

semantics, we tried to minimize the possibility of processing messages multiple
times. This means committing the message offsets to the Kafka cluster happens
right after the transaction is committed to the database.

## Configuring stream transactions

> A stream fails when a transformation fails to commit a transaction. This is a side effect
> of [isolation levels](/reference-guide/isolation-levels.md) and its behaviour can be
> configured by setting the appropriate memgraph flag:

Reviewer: It suggests me that a transaction can only fail because of conflict, but that is not true. A transaction can fail because of various reasons. I would rephrase this as:
```suggestion
A stream might fail when a transaction (in which the returned queries are executed) fails to commit because of another conflicting transaction. This is a side effect
of [isolation levels](/reference-guide/isolation-levels.md) and its behaviour can be
configured by setting the appropriate memgraph flag:
```

It is a bit rough, so feel free to rephrase it.

```
--stream-transaction-conflict-retries TIMES_TO_RETRY
```

> By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
> to 0 then memgraph tries a transaction only once. However, for any other value,
> memgraph retries with a total number of: one + `TIMES_TO_RETRY`. 

Reviewer 2: ```suggestion
By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
to 0 then memgraph tries a transaction only once. However, for any other value,
memgraph retries with a total number of: 1 + `TIMES_TO_RETRY`. 
```

> Morever, the interval of retries is also important and can be configured by 

Reviewer 2: ```suggestion
Moreover, the interval of retries is also important and can be configured by 
```

```
--stream-transaction-retry_interval INTERVAL_TIME
```

> The `INTERVAL_TIME` is meassured in `milliseconds` and its default value is `500ms`.

Reviewer 2: ```suggestion
The `INTERVAL_TIME` is measured in `milliseconds` and its default value is `500ms`.
```

semantics, we tried to minimize the possibility of processing messages multiple
times. This means committing the message offsets to the Kafka cluster happens
right after the transaction is committed to the database.

## Configuring stream transactions

A stream can fail for various reasons. One of the failures of interest is when a transaction
(in which the returned queries of the transformation are executed) fails to commit because of another conflicting 
transaction. This is a side effect of [isolation levels](/reference-guide/isolation-levels.md) 
and its behaviour can be configured by setting the appropriate memgraph flag:

```
--stream-transaction-conflict-retries=TIMES_TO_RETRY
```

> By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
> to 0 then memgraph tries a transaction only once. However, for any other value,
> memgraph retries with a total number of: 1 + `TIMES_TO_RETRY`. 

Reviewer 3: ```suggestion
By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
to 0 then memgraph tries a transaction only once. However, for any other value,
memgraph retries with a total number of: 1 + `TIMES_TO_RETRY`. 
```
Weird phrasing of it. 31 seems like a strange number.
```
Memgraph will always try to execute the transaction once. 
If it fails because of conflicts with other transactions it will retry the transaction up to `TIMES_TO_RETRY` times. 
By default, that number is set to 30....
```
